enterprise_access/apps/bffs/tests/test_response_builders.py

```diff
@@ -0,0 +1,143 @@
+            self.base_response_builder.build()
+
+    @ddt.data(
+        {
+            'errors': True,
+            'warnings': True,
+        },
+        {
+            'errors': True,
+            'warnings': False,
+        },
+        {
+            'errors': False,
+            'warnings': True,
+        },
+        {
+            'errors': False,
+            'warnings': False,
+        }
+    )
+    @ddt.unpack
+    def test_add_errors_warnings_to_response(self, errors, warnings):
+        sample_response = {
+            "some_existing_metadata": "should_still_exist"
+        }
+        expected_output = {
+            **sample_response,
+        }
+        if errors:
+            self.context.add_error(
+                **self.mock_error
+            )
+            expected_output['errors'] = [self.mock_error]
+        if warnings:
+            self.context.add_warning(
+                **self.mock_warning
+            )
+            expected_output['warnings'] = [self.mock_warning]
+        response_data = self.base_response_builder.add_errors_warnings_to_response(sample_response)
+        self.assertEqual(response_data, expected_output)
+
+    @ddt.data(
+        {
+            'status_code': 201
+        },
+        {
+            'status_code': None
+        }
+    )
+    @ddt.unpack
+    def test_get_status_code(self, status_code):
+        expected_output = status_code if status_code else 200
+        if status_code:
+            self.context.status_code = status_code
+        response_status_code = self.base_response_builder.get_status_code()
+        self.assertEqual(response_status_code, expected_output)
+
+
+@ddt.ddt
+class TestBaseLearnerResponseBuilder(TestBaseResponseBuilder, TestLicenseManagerUserApiClient):
+    def setUp(self):
+        super().setUp()
+        self.mock_enterprise_catalog_uuid_1 = self.faker.uuid4()
+        self.mock_enterprise_catalog_uuid_2 = self.faker.uuid4()
+
+        self.mock_subscription_plan_uuid_1 = self.faker.uuid4()
+        self.mock_subscription_plan_uuid_2 = self.faker.uuid4()
+
+        self.mock_customer_agreement_uuid_1 = self.faker.uuid4()
+        self.mock_customer_agreement_uuid_2 = self.faker.uuid4()
+
+        self.base_learner_response_builder = BaseLearnerResponseBuilder(self.context)
+
+    def test_common_response_logic_initialization(self):
+        response_data = self.base_learner_response_builder.common_response_logic({})
+        expected_response = {
+            'enterprise_customer_user_subsidies': {
+                'subscriptions': {}
+            }
+        }
+        self.assertEqual(response_data, expected_response)
+
+    def test_common_response_logic_subscription_context(self):
+        self.context.data['subscriptions'] = {
+            **self.mock_subscription_plan,
+            'uuid': self.mock_subscription_plan_uuid_1,
+            'enterprise_catalog_uuid': self.mock_enterprise_catalog_uuid_1,
+        }
```

Comment:
You mentioned this in Slack already, but the `subscriptions` data currently includes:

```python
{
  'customer_agreement': subscriptions_result.get('customer_agreement', {}),
  'subscription_licenses': transformed_licenses,
  'subscription_licenses_by_status': subscription_licenses_by_status,
}
```

Eventually, it'll conform to further transforms similar what exists in the frontend ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/1205/files#diff-df539c9ca1128b6cb78a470cd3c3f45470ec7e0561a02df6d8f3e1f90cb58eeaR202)).